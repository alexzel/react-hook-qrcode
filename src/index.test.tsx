import React, { ReactElement } from 'react'
import renderer from 'react-test-renderer'

import useQRCodeGenerator from './'

const Component = (
  props: {
    value: string
    level: string
    border: number
  }
): ReactElement => {
  const { value, level, border } = props
  const { parts, size } = useQRCodeGenerator(value, level)
  return (
    <svg viewBox={`0 0 ${size + border * 2} ${size + border * 2}`} stroke='none'>
      <rect width='100%' height='100%' fill='#ffffff' />
      <path d={parts.join(' ')} fill='#000000' />
    </svg>
  )
}

describe('index', () => {
  it('renders component with useQRCodeGenerator', () => {
    const component = renderer.create(
      <Component value='test' level='L' border={3} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
