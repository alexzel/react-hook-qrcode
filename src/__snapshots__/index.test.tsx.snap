// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders component with useQRCodeGenerator 1`] = `
<svg
  stroke="none"
  viewBox="0 0 27 27"
>
  <rect
    fill="#ffffff"
    height="100%"
    width="100%"
  />
  <path
    d="M0,0h1v1h-1z M1,0h1v1h-1z M2,0h1v1h-1z M3,0h1v1h-1z M4,0h1v1h-1z M5,0h1v1h-1z M6,0h1v1h-1z M8,0h1v1h-1z M9,0h1v1h-1z M10,0h1v1h-1z M12,0h1v1h-1z M14,0h1v1h-1z M15,0h1v1h-1z M16,0h1v1h-1z M17,0h1v1h-1z M18,0h1v1h-1z M19,0h1v1h-1z M20,0h1v1h-1z M0,1h1v1h-1z M6,1h1v1h-1z M9,1h1v1h-1z M10,1h1v1h-1z M14,1h1v1h-1z M20,1h1v1h-1z M0,2h1v1h-1z M2,2h1v1h-1z M3,2h1v1h-1z M4,2h1v1h-1z M6,2h1v1h-1z M10,2h1v1h-1z M14,2h1v1h-1z M16,2h1v1h-1z M17,2h1v1h-1z M18,2h1v1h-1z M20,2h1v1h-1z M0,3h1v1h-1z M2,3h1v1h-1z M3,3h1v1h-1z M4,3h1v1h-1z M6,3h1v1h-1z M8,3h1v1h-1z M11,3h1v1h-1z M12,3h1v1h-1z M14,3h1v1h-1z M16,3h1v1h-1z M17,3h1v1h-1z M18,3h1v1h-1z M20,3h1v1h-1z M0,4h1v1h-1z M2,4h1v1h-1z M3,4h1v1h-1z M4,4h1v1h-1z M6,4h1v1h-1z M10,4h1v1h-1z M11,4h1v1h-1z M14,4h1v1h-1z M16,4h1v1h-1z M17,4h1v1h-1z M18,4h1v1h-1z M20,4h1v1h-1z M0,5h1v1h-1z M6,5h1v1h-1z M9,5h1v1h-1z M11,5h1v1h-1z M14,5h1v1h-1z M20,5h1v1h-1z M0,6h1v1h-1z M1,6h1v1h-1z M2,6h1v1h-1z M3,6h1v1h-1z M4,6h1v1h-1z M5,6h1v1h-1z M6,6h1v1h-1z M8,6h1v1h-1z M10,6h1v1h-1z M12,6h1v1h-1z M14,6h1v1h-1z M15,6h1v1h-1z M16,6h1v1h-1z M17,6h1v1h-1z M18,6h1v1h-1z M19,6h1v1h-1z M20,6h1v1h-1z M9,7h1v1h-1z M12,7h1v1h-1z M2,8h1v1h-1z M4,8h1v1h-1z M5,8h1v1h-1z M6,8h1v1h-1z M8,8h1v1h-1z M9,8h1v1h-1z M13,8h1v1h-1z M17,8h1v1h-1z M20,8h1v1h-1z M3,9h1v1h-1z M4,9h1v1h-1z M7,9h1v1h-1z M8,9h1v1h-1z M12,9h1v1h-1z M14,9h1v1h-1z M15,9h1v1h-1z M19,9h1v1h-1z M20,9h1v1h-1z M0,10h1v1h-1z M1,10h1v1h-1z M2,10h1v1h-1z M3,10h1v1h-1z M4,10h1v1h-1z M6,10h1v1h-1z M8,10h1v1h-1z M12,10h1v1h-1z M13,10h1v1h-1z M14,10h1v1h-1z M15,10h1v1h-1z M17,10h1v1h-1z M18,10h1v1h-1z M19,10h1v1h-1z M20,10h1v1h-1z M0,11h1v1h-1z M1,11h1v1h-1z M2,11h1v1h-1z M3,11h1v1h-1z M5,11h1v1h-1z M8,11h1v1h-1z M9,11h1v1h-1z M11,11h1v1h-1z M12,11h1v1h-1z M15,11h1v1h-1z M16,11h1v1h-1z M19,11h1v1h-1z M0,12h1v1h-1z M1,12h1v1h-1z M5,12h1v1h-1z M6,12h1v1h-1z M8,12h1v1h-1z M9,12h1v1h-1z M10,12h1v1h-1z M15,12h1v1h-1z M17,12h1v1h-1z M19,12h1v1h-1z M20,12h1v1h-1z M8,13h1v1h-1z M10,13h1v1h-1z M11,13h1v1h-1z M12,13h1v1h-1z M17,13h1v1h-1z M20,13h1v1h-1z M0,14h1v1h-1z M1,14h1v1h-1z M2,14h1v1h-1z M3,14h1v1h-1z M4,14h1v1h-1z M5,14h1v1h-1z M6,14h1v1h-1z M9,14h1v1h-1z M10,14h1v1h-1z M16,14h1v1h-1z M17,14h1v1h-1z M19,14h1v1h-1z M20,14h1v1h-1z M0,15h1v1h-1z M6,15h1v1h-1z M8,15h1v1h-1z M9,15h1v1h-1z M10,15h1v1h-1z M12,15h1v1h-1z M14,15h1v1h-1z M19,15h1v1h-1z M0,16h1v1h-1z M2,16h1v1h-1z M3,16h1v1h-1z M4,16h1v1h-1z M6,16h1v1h-1z M8,16h1v1h-1z M10,16h1v1h-1z M11,16h1v1h-1z M14,16h1v1h-1z M16,16h1v1h-1z M17,16h1v1h-1z M19,16h1v1h-1z M20,16h1v1h-1z M0,17h1v1h-1z M2,17h1v1h-1z M3,17h1v1h-1z M4,17h1v1h-1z M6,17h1v1h-1z M9,17h1v1h-1z M14,17h1v1h-1z M15,17h1v1h-1z M19,17h1v1h-1z M0,18h1v1h-1z M2,18h1v1h-1z M3,18h1v1h-1z M4,18h1v1h-1z M6,18h1v1h-1z M8,18h1v1h-1z M12,18h1v1h-1z M13,18h1v1h-1z M15,18h1v1h-1z M18,18h1v1h-1z M20,18h1v1h-1z M0,19h1v1h-1z M6,19h1v1h-1z M9,19h1v1h-1z M10,19h1v1h-1z M11,19h1v1h-1z M13,19h1v1h-1z M14,19h1v1h-1z M15,19h1v1h-1z M16,19h1v1h-1z M17,19h1v1h-1z M19,19h1v1h-1z M0,20h1v1h-1z M1,20h1v1h-1z M2,20h1v1h-1z M3,20h1v1h-1z M4,20h1v1h-1z M5,20h1v1h-1z M6,20h1v1h-1z M10,20h1v1h-1z M13,20h1v1h-1z M14,20h1v1h-1z M15,20h1v1h-1z M18,20h1v1h-1z M19,20h1v1h-1z M20,20h1v1h-1z"
    fill="#000000"
  />
</svg>
`;
